core/runtime/src/main/java/io/quarkus/runtime/configuration/MemorySize.java

```diff
@@ -0,0 +1,28 @@
+package io.quarkus.runtime.configuration;
+
+import java.math.BigInteger;
+
+/**
+ * A type representing data sizes.
+ */
+public class MemorySize {
+    private BigInteger value;
+
+    public MemorySize(BigInteger value) {
+        this.value = value;
+    }
+
+    /**
+     * @return {@link Long} - the size of memory in bytes
+     */
+    public long asLongValue() {
+        return value.longValue();
+    }
+
+    /**
+     * @return {@link BigInteger} - the value
+     */
+    public BigInteger getValue() {
```

Comment:
This should be `asBigInteger` because the fact that this is a `BigInteger` internally is an implementation detail.